flink-libraries/flink-table/src/main/scala/org/apache/flink/table/descriptors/PrimitiveTypeDescriptor.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.descriptors
+
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo
+import org.apache.flink.table.typeutils.TypeStringUtils
+
+/**
+  * Descriptor for a primitive type.
+  */
+class PrimitiveTypeDescriptor[T] extends HierarchyDescriptor {
+
+  // TODO not sure if we should the BasicTypeInfo here
+  var typeInformation: BasicTypeInfo[T] = _
+  var value: T = _
+
+  def setType(basicType: BasicTypeInfo[T]): PrimitiveTypeDescriptor[T] = {
```

Comment:
This should be regular type information. The `TypeStringUtils` can stringify every valid type information.

Reply:
Yes, the `TypeStringUtils` can extract every valid type information and the user can specify the type via the API (e.g., `setType(Types.SHORT)`). I just wonder how the basic types supported by Java can be properly inferred from the config file (e.g., how to decide a parameter `1` is a byte, a short or an int). Although a short value or a byte value can be represented with an int, that will affect the constructor searching via Java reflection. Do you have any ideas for that?

Reply:
We should only support very simple type inference logic for values without specified types. E.g.:
- "1" -> always int
- "1.0" -> always double
- "true/false" -> always boolean
- string otherwise

This covers 80% of all parameters. For everything else, declare a type and we will call `Byte.valueOf()` etc.

Reply:
Sounds reasonable. Thanks.